cosmwasm-js/packages/sdk/src/wallet.ts

```diff
@@ -0,0 +1,24 @@
+import {StdSignature} from "./types";
+
+export type PrehashType = "sha256" | "sha512" | null;
+
+export type Algo = "secp256k1" | "ed25519" | "sr25519";
+
+export interface AccountData {
+  // bech32-encoded
+  readonly address: string;
+  readonly algo: Algo;
+  readonly pubkey: Uint8Array;
+}
+
+export interface OfflineSigner {
+  /**
+   * Get AccountData array from wallet. Rejects if not enabled.
+   */
+  readonly getAccounts: () => Promise<readonly AccountData[]>;
+
+  /**
+   * Request signature from whichever key corresponds to provided bech32-encoded address. Rejects if not enabled.
+   */
+  readonly sign: (address: string, message: Uint8Array, prehashType?: PrehashType) => Promise<StdSignature>;
```

Comment:
Can you change the sign method interface as [readonly sign: (signerAddress: string, signDoc: StdSignDoc) => Promise<SignResponse>](https://github.com/cosmos/cosmjs/blob/v0.23.1/packages/launchpad/src/signer.ts#L37) as this is more recent version that Keplr supports.

Reply:
Sorry, I totally missed this. I'll work to address your suggestions.

BTW This PR works very good for us in [this code segment](https://github.com/enigmampc/EthereumBridgeFrontend/blob/7c996239160863d278830e0ae3059020cb9c55ba/src/stores/UserStore.ts#L125-L136) with Keplr's `OfflineSigner`. Is this Okay?

Reply:
Done. I made the `SigningCosmWasmClient` constructor a bit different from what you suggested, because if you pass `enigmautils` then `seed` is irrelevant, so now you can pass one or the other, but not both.

Look at the difference from your suggestion and let me know what you think ([Link to code](https://github.com/enigmampc/SecretNetwork/blob/3f4a7d6038303200c1cc6c39f412d7e2f4bddb43/cosmwasm-js/packages/sdk/src/signingcosmwasmclient.ts#L123-L145)):
```diff
public constructor(
     apiUrl: string,
     senderAddress: string,
     signer: SigningCallback | OfflineSigner,
-    enigmautils?: SecretUtils,
-    seed?: Uint8Array,
+    seedOrEnigmaUtils?: Uint8Array | SecretUtils,
     customFees?: Partial<FeeTable>,
     broadcastMode = BroadcastMode.Block,
   ) {
-    super(apiUrl, seed, broadcastMode);
+    if (seedOrEnigmaUtils instanceof Uint8Array) {
+      super(apiUrl, seedOrEnigmaUtils, broadcastMode);
+    } else {
+      super(apiUrl, undefined, broadcastMode);
+    }
+
     this.anyValidAddress = senderAddress;
     this.senderAddress = senderAddress;
     //this.signCallback = signCallback ? signCallback : undefined;
     this.signer = signer;
-    if (enigmautils) {
-      this.restClient.enigmautils = enigmautils;
+    if (seedOrEnigmaUtils && !(seedOrEnigmaUtils instanceof Uint8Array)) {
+      this.restClient.enigmautils = seedOrEnigmaUtils;
     }
```

Reply:
Looks good to me

Reply:
Thanks, this is already on NPM as `secretjs@0.10.2`